crates/orderbook/src/api/order_validation.rs

```diff
@@ -376,6 +377,19 @@ impl OrderValidating for OrderValidator {
             .await
         {
             Ok(_) => Ok((order, unsubsidized_fee)),
+            Err(
+                TransferSimulationError::InsufficientAllowance
+                | TransferSimulationError::InsufficientBalance,
+            ) if order.order_creation.signature.scheme() == SigningScheme::PreSign => {
+                // We have an exception for pre-sign orders where they do not
+                // require sufficient balance or allowance. The idea, is that
+                // this allows smart contracts to place orders bundled with
+                // other transactions that either produce the required balance
+                // or set the allowance. This would, for example, allow a Gnosis
+                // Safe to bundle the pre-signature transaction with a WETH wrap
+                // and WETH approval to the vault relayer contract.
```

Comment:
Very helpful comment! Without it this exception to the rule could seem arbitrary.